### DISCUSSION TITLE
Assign new transcripts to new loci

### DISCUSSION THREAD

**Discussion by francicco** [ConesaLab/SQANTI3#297] [General] [answered]
Hi,

I'm updating the annotation for a non-model species in preparation for some scRNA data. I've got ONT data on strand-specific isoforms of the tissue of interest. Now I'm using a combination of tools to fix and update my current annotation. What I'm doing is the following:

1. Map the fastq reads with minimap2
2. Run IsoQuant to predict isoforms and new loci using the BAM file
3. Run SQUANTI3: `sqanti3_qc.py` and `sqanti3_filter.py` to filter out the erroneous transcript.

The whole thing seems to work well just using 10% of ONT reads, and on a scRNAseq test data, I go from 14% of reads mapped in intergenic regions to only 4%. 

The only problem is that new transcripts are not assigned to existing or new loci. Is there anything wrong I'm doing what do you think I should do?

Thanks a lot
Francesco

**Comment by carolinamonzo** [comment]
Hi @francicco I'm not sure what you mean by new transcripts not assigned to existing or new loci. Could you give us an example of what you are encountering?

**Reply by francicco** [reply]
Hi @carolinamonzo, 

I can explain my aim better, quickly.

I'm preparing (updating/fixing) an annotation I have (non-model species) for a bunch of libraries of single-cell experiments. My annotation suffers from having incomplete/short UTR regions. I want to fix this using dRNA ONT reads. The mapping seems to be quite good, especially after some filtering (quality/too-long introns probably due to TEs).

So what I'm doing now is reconstructing transcripts which then I would like to filter with SQANTI3. Comparing at least 3 different methods (StringTie, IsoQuant, and Flair), it's obvious that their union is the best option. Each one of them is not very good in assigning transcripts to loci, mostly because I'm running them **without** a guided annotation otherwise they won't annotate UTRs in case transcripts have the same intron chain. At least this is what I observe.

I actually have a lot of question about SQANTI3, but specifically for this thread, I was wondering if SQANTI is able to cluster these transcripts into loci, a bit like Mikado can do.

Currently, the reason why I'm hesitant in using Mikado, it's because it's a bit harsh with these annotation and might reject a lot of them and a scoring exploration seems to be required.

I hope I explained my problem clear.

This is how the data looks like, specifically for StringTie and different steps with SQANTI3.
<img width="2343" alt="Screenshot 2024-05-24 at 10 50 44 AM" src="https://github.com/ConesaLab/SQANTI3/assets/9006870/4debc6fb-a821-43aa-95ce-9b68b479831d">

Thanks a lot
Francesco

**Comment by carolinamonzo** [comment, marked as answer]
Hi @francicco,
Sorry it took so long to get back to you. SQANTI does not cluster transcripts into loci. What we usually do to join annotations from different transcript reconstruction pipelines is either use TAMA merge (https://github.com/GenomeRIK/tama), or sumarize transcript models into their unique junction chains, and use that as the input to SQANTI QC and SQANTI filter to check the veracity of those transcripts.
